luigi/worker.py

```diff
@@ -678,6 +707,26 @@ def _log_remote_tasks(self, running_tasks, n_pending_tasks, n_unique_pending):
             if n_unique_pending:
                 logger.debug("There are %i pending tasks unique to this worker", n_unique_pending)
 
+    def _get_work_task_id(self, get_work_response):
+        if get_work_response['task_id'] is not None:
+            return get_work_response['task_id']
+        elif 'batch_id' in get_work_response:
+            task = load_task(
+                module=get_work_response.get('task_module'),
+                task_name=get_work_response['task_family'],
+                params_str=get_work_response['task_params'],
+            )
+            self._scheduler.add_task(
+                worker=self._id,
+                task_id=task.task_id,
+                module=get_work_response.get('task_module'),
+                family=get_work_response['task_family'],
+                params=task.to_str_params(),
+                status=RUNNING,
+                batch_id=get_work_response['batch_id'],
+            )
```

Comment:
Wait, this `add_task` worries me a bit. Previously a worker have not needed to say "I'm RUNNING TaskABC()", because the scheduler have already told the worker to run a task. Why do things need to be different for the BATCH case?

I'm not sure how useful the invariant of `get_work` is never called with `status=RUNNING` actually is. But I'm interested to hear why it can't be kept. :)